app/services/projects/creation_wizard/reupload_artifact_on_status_changes_service.rb

```diff
@@ -0,0 +1,150 @@
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Projects::CreationWizard
+  class ReuploadArtifactOnStatusChangesService
+    include Contracted
+    include ProjectHelper
+    include Rails.application.routes.url_helpers
+    prepend Projects::Concerns::UpdateDemoData
+
+    attr_reader :current_user, :work_package
+
+    delegate :project, to: :work_package
+
+    def initialize(current_user:, work_package:)
+      @current_user = current_user
+      @work_package = work_package
+    end
+
+    def call!(changes:)
+      return unless OpenProject::FeatureDecisions.project_initiation_active?
+      return if changes["status_id"].blank?
+      return unless update_is_artifact_work_package?
+
+      User.execute_as_admin(current_user) do
+        update_artifact
+      end
+    end
+
+    def update_is_artifact_work_package?
+      work_package.project.project_creation_wizard_artifact_work_package_id.to_s == work_package.id.to_s
+    end
+
+    private
+
+    def update_artifact
+      call = store_artifact
+      if call.success?
+        Rails.logger.debug { "Updated artifact for creation wizard in ##{work_package.id}" }
+      else
+        Rails.logger.error("Failed to process artifact change for work package ##{work_package.id}: ##{call.message}")
+      end
+    end
+
+    def store_artifact
+      if store_attachment_locally?
+        return add_attachment_locally
+      end
+
+      if project_storage.nil?
+        return ServiceResult.failure(message: I18n.t("projects.wizard.create_artifact_storage_error"))
+      end
+
+      upload_artifact_to_storage
+    end
+
+    def store_attachment_locally?
+      project.project_creation_wizard_artifact_export_type == "attachment"
+    end
+
+    def upload_artifact_to_storage
+      export = create_pdf_export!
+
+      Storages::UploadFileService
+        .call(
+          container: work_package,
+          project_storage:,
+          file_path: project.project_creation_wizard_artifact_name,
+          file_data: StringIO.new(export.content),
+          filename: export.title
+        )
+    end
+
+    def project_storage
+      return @project_storage if defined?(@project_storage)
+
+      @project_storage = project
+        .project_storages
+        .find_by(id: project.project_creation_wizard_artifact_export_storage)
```

Comment:
```suggestion
      @project_storage ||= project
        .project_storages
        .find_by(id: project.project_creation_wizard_artifact_export_storage)
```

no need for the extra early return

Reply:
That's actually caused by rubocop, not to use find_by with memoize, as it will retrigger if nil